code/datums/traits/negative.dm

```diff
@@ -228,6 +161,306 @@
 /datum/quirk/family_heirloom/on_clone(data)
 	heirloom = data
 
+/datum/quirk/family_heirloom/proc/itemlist_species(var/mob/living/carbon/human/H)
+	var/list/heirloom_addition = list()
+
+	switch(H.dna.species.id)
+		if(SPECIES_HUMAN)
+			heirloom_addition += /obj/item/clothing/head/kitty //you had terrible parents.
+		if(SPECIES_FELINID)
+			heirloom_addition += /obj/item/clothing/head/kitty
+		if(SPECIES_LIZARD)
+			heirloom_addition += /obj/item/toy/plush/lizardplushie
+		if(SPECIES_OOZELING)
+			heirloom_addition += /obj/item/toy/plush/slimeplushie
+		if(SPECIES_MOTH)
+			heirloom_addition += /obj/item/flashlight/lantern/heirloom_moth
+			heirloom_addition += /obj/item/toy/plush/moth
+		if(SPECIES_APID)
+			heirloom_addition += /obj/item/toy/plush/beeplushie
+		if(SPECIES_PLASMAMAN)
+			heirloom_addition += /obj/item/coin/plasma
+		if(SPECIES_ETHEREAL)
+			heirloom_addition += /obj/item/coin/plasma //I am not sure what to give them
+		if(SPECIES_IPC)
+			heirloom_addition += /obj/item/disk/data
+		//retired beecode speices, but let's give them some love
+		if(SPECIES_FLY)
+			heirloom_addition += /obj/item/reagent_containers/food/drinks/bottle/virusfood
+		else
+			heirloom_addition += /obj/item/toy/eldrich_book //spooky, eldritch
+
+	// You get everything for 1% chance.
+	if(prob(1))
+		heirloom_addition += /obj/item/clothing/head/kitty
+		heirloom_addition += /obj/item/toy/plush/lizardplushie
+		heirloom_addition += /obj/item/toy/plush/slimeplushie
+		heirloom_addition += /obj/item/flashlight/lantern/heirloom_moth
+		heirloom_addition += /obj/item/toy/plush/moth
+		heirloom_addition += /obj/item/toy/plush/beeplushie
+		heirloom_addition += /obj/item/coin/plasma
+		heirloom_addition += /obj/item/disk/data
+		heirloom_addition += /obj/item/reagent_containers/food/drinks/bottle/virusfood
+		heirloom_addition += /obj/item/toy/eldrich_book
+
+	return heirloom_addition
+
+/datum/quirk/family_heirloom/proc/itemlist_jobs()
+	var/list/heirloom_addition = list()
+	var/DEPARTMENT_FLAG = 0 //For department shared items
+
+	// Job specific table
+	switch(quirk_holder.mind.assigned_role)
+		//Service jobs
+		if(JOB_STRING_ASSISTANT)
+			DEPARTMENT_FLAG |= DEPARTMENT_SERVICE
+			heirloom_addition += /obj/item/storage/toolbox/mechanical/old/heirloom
```

Comment:
instead of doing += 10 times just add list to list

Reply:
the datum is fully changed. thanks.